mteb/tasks/Retrieval/multilingual/StatcanDialogueDatasetRetrieval.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+import datasets
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks import MultilingualTask
+from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+
+_EVAL_SPLITS = ["dev", "test"]
+
+_LANGS = {
+    # <iso_639_3>-<ISO_15924>
+    "english": ["eng-Latn"],
+    "french": ["fra-Latn"],
+}
+
+
+def _load_statcan_data(
+    path: str, langs: list, splits: str, cache_dir: str = None, revision: str = None
+):
+    queries = {lang: {split: {} for split in splits} for lang in langs}
+    corpus = {lang: {split: {} for split in splits} for lang in langs}
+    relevant_docs = {lang: {split: {} for split in splits} for lang in langs}
+
+    for split in splits:
+        for lang in langs:
+            query_table = datasets.load_dataset(
+                path,
+                f"queries_{lang}",
+                split=split,
+                cache_dir=cache_dir,
+                revision=revision,
+            )
+            corpus_table = datasets.load_dataset(
+                path,
+                "corpus",
+                split=lang,
+                cache_dir=cache_dir,
+                revision=revision,
+            )
+
+            for row in query_table:
+                query = row["query"]
+                query_id = row["query_id"]
+                doc_id = row["doc_id"]
+                queries[lang][split][query_id] = query
+                if query_id not in relevant_docs[lang][split]:
+                    relevant_docs[lang][split][query_id] = {}
+                relevant_docs[lang][split][query_id][doc_id] = 1
+
+            for row in corpus_table:
+                doc_id = row["doc_id"]
+                doc_content = row["doc"]
+                corpus[lang][split][doc_id] = {"text": doc_content}
+
+    corpus = datasets.DatasetDict(corpus)
+    queries = datasets.DatasetDict(queries)
+    relevant_docs = datasets.DatasetDict(relevant_docs)
+
+    return corpus, queries, relevant_docs
+
+
+class StatcanDialogueDatasetRetrieval(MultilingualTask, AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="StatcanDialogueDatasetRetrieval",
+        description="A Dataset for Retrieving Data Tables through Conversations with Genuine Intents, available in English and French.",
+        dataset={
+            "path": "McGill-NLP/statcan-dialogue-dataset-retrieval",
+            "revision": "main",
+        },
+        type="Retrieval",
+        category=None,  # TBD
+        eval_splits=_EVAL_SPLITS,
+        eval_langs=_LANGS,
+        main_score="recall_at_10",
```

Comment:
Does StatsCan have multiple matching gold documents/table or single?

Reply:
Single document per example

Reply:
Sorry, there might be more than 1 in some cases.